packages/labs/analyzer/src/lib/javascript/classes.ts

```diff
@@ -223,13 +223,61 @@ export const getHeritage = (
   };
 };
 
+const getIdentifierFromMixin = (
+  expression: ts.CallExpression,
+  analyzer: AnalyzerInterface
+): ts.Identifier | null => {
+  const [firstArg] = expression.arguments;
+
+  if (expression.arguments.length !== 1 || !firstArg) {
+    return null;
+  }
+
+  if (analyzer.typescript.isIdentifier(firstArg)) {
+    return firstArg;
+  }
+
+  if (analyzer.typescript.isCallExpression(firstArg)) {
+    return getIdentifierFromMixin(firstArg, analyzer);
+  }
+
+  return null;
+};
+
+const getMixinsFromExpression = (
```

Comment:
im thinking these two functions might be better as one which returns a `ClassHeritage` or something, so we traverse only once and get the mixins/superclass back in one go